Simplenote/RemoteError.swift

```diff
@@ -1,38 +1,50 @@
 import Foundation
 
-enum RemoteError: Error {
-    case network
-    case tooManyAttempts
-    case requestError(Int, Error?)
+
+// MARK: - RemoteError
+//
+struct RemoteError: Error {
+    let statusCode: Int
+    let response: String?
+    let networkError: Error?
```

Comment:
We're making a couple changes, to simplify things here:

1. Whenever `statusCode` is nil, we'll map that into `statusCode = 0`
2. `statusCode = 0` implies networking error. Up to the next layer to map
3. `tooManyAttempts` will also be picked up by the next layer

All of these changes allow us to convert `RemoteError` into a Struct, instead